libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -3258,68 +3258,52 @@ string YulUtilFunctions::conversionFunction(Type const& _from, Type const& _to)
 		switch (fromCategory)
 		{
 		case Type::Category::Address:
+		case Type::Category::Contract:
 			body =
 				Whiskers("converted := <convert>(value)")
 					("convert", conversionFunction(IntegerType(160), _to))
 					.render();
 			break;
 		case Type::Category::Integer:
 		case Type::Category::RationalNumber:
-		case Type::Category::Contract:
 		{
+			solAssert(_from.mobileType(), "");
 			if (RationalNumberType const* rational = dynamic_cast<RationalNumberType const*>(&_from))
-				solUnimplementedAssert(!rational->isFractional(), "Not yet implemented - FixedPointType.");
+				if (rational->isFractional())
+					solAssert(toCategory == Type::Category::FixedPoint, "");
+
 			if (toCategory == Type::Category::FixedBytes)
 			{
-				solAssert(
-					fromCategory == Type::Category::Integer || fromCategory == Type::Category::RationalNumber,
-					"Invalid conversion to FixedBytesType requested."
-				);
 				FixedBytesType const& toBytesType = dynamic_cast<FixedBytesType const&>(_to);
 				body =
 					Whiskers("converted := <shiftLeft>(<clean>(value))")
-						("shiftLeft", shiftLeftFunction(256 - toBytesType.numBytes() * 8))
-						("clean", cleanupFunction(_from))
-						.render();
+					("shiftLeft", shiftLeftFunction(256 - toBytesType.numBytes() * 8))
+					("clean", cleanupFunction(_from))
+					.render();
 			}
 			else if (toCategory == Type::Category::Enum)
-			{
-				solAssert(_from.mobileType(), "");
 				body =
 					Whiskers("converted := <cleanEnum>(<cleanInt>(value))")
 					("cleanEnum", cleanupFunction(_to))
-					// "mobileType()" returns integer type for rational
```

Comment:
I don't see why we would need cleanup here for rational numbers, it is actually better if the integer stays large so that it will be caught by the enum cleanup.